src/main/java/com/example/employeecrudapplication/validator/EmployeeValidator.java

```diff
@@ -0,0 +1,45 @@
+package com.example.employeecrudapplication.validator;
+
+import com.example.employeecrudapplication.exception.EmployeeValidationException;
+import com.example.employeecrudapplication.model.EmployeeDto;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EmployeeValidator {
+
+    public void validateEmail(String email) {
+        if (
```

Comment:
Make normal email validation (e.g., regex)

Reply:
Changed validateEmail and created a method patternMatches (is it a helper method?)
![image](https://user-images.githubusercontent.com/100875409/219571221-1a7516ef-9ee9-42fd-89f6-b183836fcef2.png)

Reply:
1. Variable at line 25 is redundant
2. You call method _patternMatches_ twice (line 28 and 29)
3. You could make patternMatches method private and non-static

Reply:
1. Got rid of the variable
2. Removed the unnecessary method call
3. I've made patternMatches private and non-static, but now I don't understand how to use .patternMatches. The IDE suggests that I could make it static to solve this problem. What should I do?

Reply:
Call that method without **EmployeeValidator.**, just like plain method

Reply:
Done
![image](https://user-images.githubusercontent.com/100875409/219578832-b5edc5c4-2d73-46c8-a163-f5f0a949f758.png)

Reply:
To follow the best practices, move the private method to the bottom of this class